enhancements/node-latency-profile/node-latency-profiles.md

```diff
@@ -0,0 +1,354 @@
+
+How will the component handle version skew with other components?
+What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- During an upgrade, we will always have skew among components, how will this impact your work?
+
+  This functionality only modifies the existing arguments of the `Kubelet`, `Controller Manager` and `Kube API Server`. As long as components keep the respective flags in place, version skew should not have any impact on this work.
+
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+
+  Yes, It does involve coordinating between the control plane and the kubelet. N-2 kubelet without this feature will still work because the kubelet argument we are using exists from beyond N-2.
+
+- Will any other components on the node change? For example, changes to CSI, CRI
+  or CNI may require updating that component before the kubelet.
+
+  No
+
+
+
+### Risks and Mitigations
+
+1. Any bug in the coordination between the `Kubelet`, `Controller Manager` and `Kube API Server` might put the cluster at risk.
+
+2. If the rollback mechanism proposed in this enhancement also fails for some reason, then users can mitigate manually supplying `KubeletConfig` that could work with the `Controller Manager` and `Kube API Server`.
+
+3. We are changing `--node-monitor-period` and/or `--node-monitor-grace-period` arguments of the Controller Manager and `--default-not-ready-toleration-seconds` and `--default-unreachable-toleration-seconds` of arguments the Kube API Server.
+Although we are targeting only worker nodes with this enhancement, both Controller Manager and Kube API Server do not allow us to set those respective arguments only for the subset of the available nodes.
+i.e. They are applied cluster wide including master nodes. We are only proposing `WorkerLatencyProfile` with this enhancement that slow down updates from the Kubelet on the worker nodes and the corresponding reaction time from the control plane.
+So even though 100% reliable network connections amongst master nodes is assumed, when the user applies `WorkerLatencyProfile` there will be delays evicting the pods running on the master nodes than their usual default values.
+
+### Upgrade / Downgrade Strategy
+
+Since this feature is controlled using the `KubeletConfig` and `ConfigObserver`, upgrade/downgrade strategies applicable for the `KubeletConfig` and `ConfigObserver` are applicable here too.
+
+## Drawbacks
+
+This solution is fairly complex due to the synchronization it requires between the control plane and the kubelet. Unfortunately, there isn't any way around this.
+
+## Alternatives
+
+Since we don't want users to manually modify the arguments for the components involved, there isn't any viable alternative that could safely work.
```

Comment:
I see alternatives for the API. Why is it not part of the to-be-introduced Node object?